modules/gateway/peers.go

```diff
@@ -442,14 +442,9 @@ func (g *Gateway) managedConnectNewPeer(conn net.Conn, remoteVersion string, rem
 	return nil
 }
 
-// Connect establishes a persistent connection to a peer, and adds it to the
-// Gateway's peer list.
-func (g *Gateway) Connect(addr modules.NetAddress) error {
-	if err := g.threads.Add(); err != nil {
-		return err
-	}
-	defer g.threads.Done()
-
+// managedConnect establishes a persistent connection to a peer, and adds it to
+// the Gateway's peer list.
+func (g *Gateway) managedConnect(addr modules.NetAddress) error {
```

Comment:
Ah okay. I was thinking you could get rid of `tg.Add(); defer tg.Done()` in the code below and instead call `Connect`. But then the call to `g.log.Debugln` could panic when the gateway is closing. 

For reference here is the relevant code where `managedConnect` is called in peers.go.
```go
 +		// Try connecting to that peer in a goroutine.
 +		go func() {
 +			if err := g.threads.Add(); err != nil {
 +				return
 +			}
 +			defer g.threads.Done()
 +
 +			err := g.managedConnect(addr)
 +			if err != nil {
 +				g.log.Debugln("WARN: automatic connect failed:", err)
 +			}
 +		}()
```